Jetpack Cloud: Update the design of Jetpack Scan section based on the latest styles

rcanepa opened the pull request:
#### Changes proposed in this Pull Request

- Make the Jetpack Scan section match designs from our Figma file.

See 1151678672052943-as-1170456139910593.

#### Testing instructions

##### Okay state

- Navigate to http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=okay
- Make sure styles match our Figma file

##### Scanning state

- Navigate to http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=scanning
- Make sure styles match our Figma file

##### Threats state

- Navigate to http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=threats
- Make sure styles match our Figma file

##### Error state

- Navigate to http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=error
- Make sure styles match our Figma file

#### Screenshots
![image](https://user-images.githubusercontent.com/3418513/78928696-758b9400-7a77-11ea-99b6-a72317efa851.png)
![image](https://user-images.githubusercontent.com/3418513/78928733-820fec80-7a77-11ea-990f-1e1eedca58c0.png)
![image](https://user-images.githubusercontent.com/3418513/78928761-8b00be00-7a77-11ea-9aba-4c077dd56bea.png)
![image](https://user-images.githubusercontent.com/3418513/78928784-981dad00-7a77-11ea-83a7-4eeb2517b2b9.png)
![image](https://user-images.githubusercontent.com/3418513/78928799-a23fab80-7a77-11ea-825d-640a6ecf1ff7.png)

enejb commented on client/landing/jetpack-cloud/components/scan-history-item/style.scss:
```diff
@@ -41,7 +47,7 @@
 	}
 
-	&__badge {
+	&__badge.badge {
```
I noticed that this badge is not quite centered in height. 
Updating the line-height from 17 -> 14px fixed that for me. 

Before:
<img width="109" alt="Screen Shot 2020-04-10 at 10 16 55 AM" src="https://user-images.githubusercontent.com/115071/78975460-6ad30c80-7b14-11ea-935a-1535ac1286a9.png">

After:
<img width="100" alt="Screen Shot 2020-04-10 at 10 16 35 AM" src="https://user-images.githubusercontent.com/115071/78975432-5f7fe100-7b14-11ea-999a-46e83226cc34.png">

```suggestion
	&__badge.badge {
	line-height: 14px;
```

enejb commented:
Nice work here. 

I noticed a couple of thing. 
1. The Fix all button turns gray on hover.
See:
<img width="283" alt="Screen Shot 2020-04-10 at 10 20 42 AM" src="https://user-images.githubusercontent.com/115071/78975848-504d6300-7b15-11ea-86d7-a110b3476181.png">

And the minor thing with the badge not being vertically centered. 
Otherwise I think this looks really good. 

I think it would be good to make some of the changes that are here in this PR more global.
But I think that could be done in future PRs. :)

rcanepa commented:
I pushed two commits that fix:

- Badge vertical alignment.
- Fix all button styles.

Demo:
![image](https://user-images.githubusercontent.com/3418513/79003952-189ce600-7b2a-11ea-93d1-2db91ad5a9dd.png)

![Kapture 2020-04-10 at 12 51 34](https://user-images.githubusercontent.com/3418513/79003936-0f137e00-7b2a-11ea-8de4-73be3e6b6f47.gif)

keoshi commented:
The buttons look much better now!

Though I guess this had some unintentional side-effects:

![image](https://user-images.githubusercontent.com/390760/79117160-52682980-7d82-11ea-9c1e-70915693b4b9.png)

![image](https://user-images.githubusercontent.com/390760/79117172-58f6a100-7d82-11ea-9327-ed30454c1c6e.png)

![image](https://user-images.githubusercontent.com/390760/79117211-6b70da80-7d82-11ea-982d-e0381c09d7e1.png)

We probably need to restrict these styles to `.main` or `.layout__primary`, and add styles/classes for “invisible” buttons (which wouldn't have a background or borders).

Finally, I agree with you on the location — maybe we should create a JPC-global `style.scss` in `client/landing/jetpack-cloud` and use it for common styles?

Thanks for making those changes, appreciate it!

rcanepa commented:
Hey @keoshi !

> We probably need to restrict these styles to .main or .layout__primary, and add styles/classes for “invisible” buttons (which wouldn't have a background or borders).

Sure. The idea is to avoid side-effects on the side bar and other components?

> Finally, I agree with you on the location — maybe we should create a JPC-global style.scss in client/landing/jetpack-cloud and use it for common styles?

Great suggestion. I'll move the styles to that location.

Two questions:

a) The Jetpack Cloud Style Guide doesn't include a button like this one:
![image](https://user-images.githubusercontent.com/3418513/79123281-2cd22480-7d70-11ea-8eaf-06b2d1fe214a.png)

It resembles the secondary but it has a green border. Is this an exception or could it appear on other places as well? I wonder if we should add styles for this button in the new stylesheet we are creating.

b) What is an invisible button? Do you have an example?

keoshi commented:
@rcanepa Thanks for noticing that, we hadn't so kudos for bringing it up. A bit late for sure, but we've ran an impromptu poll on p1586793350312400-slack-jetpack-design and the majority agrees that we should follow the pattern existing on the Scan mocks.

I've updated the [Figma Style Guide file](https://www.figma.com/file/r9c7MPvvxT8QLLP8qr25Hc/Jetpack-Cloud-Style-Guide?node-id=923%3A1891&viewport=-3803%2C388%2C1) and will be jumping into this PR tomorrow morning to push those changes.

Thanks again!

keoshi commented:
> The idea is to avoid side-effects on the side bar and other components?

Yes. Do you think it's a bit severe to nest them under `.main` for example? Since all of our content/visuals live in that section, it would be an easier way to style everything knowing it won't bleed over.

keoshi commented:
Quick update @rcanepa: pushed a couple of changes to this PR, let me know if they look good to you.

![2020-04-13 18 11 22](https://user-images.githubusercontent.com/390760/79141913-9a06a980-7db2-11ea-92e1-552c947f19d9.gif)

rcanepa commented:
I looks much nicer now! Thanks for adding those changes.

> I've updated the Figma Style Guide file and will be jumping into this PR tomorrow morning to push those changes.

Got it!

> Yes. Do you think it's a bit severe to nest them under .main for example? Since all of our content/visuals live in that section, it would be an easier way to style everything knowing it won't bleed over.

Yup. I'll change that. However, I noticed one problem with this approach. Elements inside modals are outside `.main`, which means that we can't target them with `.theme-jetpack-cloud .main` (the same happens with `.layout__primary`). This is how the server credential form buttons look like:

![image](https://user-images.githubusercontent.com/3418513/79146499-e349ff80-7d98-11ea-99a6-95c9bacc9112.png)

To fix that, I added the `.theme-jetpack-cloud .ReactModalPortal` selector. Let me know what you think!

enejb commented:
In my testing this worked really well. 
Nicely done @rcanepa and @keoshi

keoshi commented:
> To fix that, I added the `.theme-jetpack-cloud .ReactModalPortal` selector. Let me know what you think!

Nice, @rcanepa! Sounds like a solid approach, we just have to keep an eye for elements that are outside of these two, but I imagine there shouldn't be any in the near future.
